### PR TITLE
Don't include version in soname ELF header if versioning is disabled

### PR DESCRIPTION
**Pull request by sdroege**
And the equivalent change for macOS/iOS.

Fixes https://github.com/lu-zero/cargo-c/issues/295

### CODE REVIEW

File: src/target.rs [hunk 1]
```diff
@@ -78,15 +78,27 @@ impl Target {
             || os == "haiku"
```
Comment (sdroege) [inline src/target.rs:78]: Some lines above, why do we never set a version in the soname for Android @lu-zero ?

File: src/target.rs [hunk 2]
```diff
@@ -78,15 +78,27 @@ impl Target {
...
+                    format!("{}.{}", major, minor)
+                } else {
+                    format!("{}", major)
+                };
+                format!("-Wl,-install_name,{1}/lib{0}.{5}.dylib,-current_version,{2}.{3}.{4},-compatibility_version,{5}",
+                        lib_name, libdir.display(), major, minor, patch, install_ver)
             } else {
-                format!("{}", major)
+                format!(
+                    "-Wl,-install_name,{1}/lib{0}.dylib",
```
Comment (sdroege) [inline src/target.rs:97]: @nirbheek Does this match what meson is doing on macOS?

File: src/target.rs (hunk 1 above)
Comment (lu-zero) [reply src/target.rs:78]: because android does not have the version it its libraries.

Comment (sdroege) [reply src/target.rs:78]: First time I hear that :) Do you have some more details about that?

Comment (lu-zero) [reply src/target.rs:78]: It is a well known annoyance, you may look at stackoverflow or the hopefully now closed bugs at common software such openssl ^^; I cannot find a document that explicitly mentions it but that's how it is, assuming nothing changed in the past year.

**Comment by stephan57160** [comment]
Thx all.

Any idea when this will come "official" ?

File: src/target.rs (hunk 1 above)
Comment (sdroege) [reply src/target.rs:78]: How strange :) Thanks, I learned something new.

Comment (stephan57160) [reply src/target.rs:78]: @lu-zero: This is still true with the last NDKs. 
does not have and does not want !

File: src/target.rs (hunk 2 above)
Comment (nirbheek) [reply src/target.rs:97]: Yes, it's close enough to that.

On macOS, Meson allows you to set `current_version` and `compatibility_version` specifically with the [`darwin_versions:` kwarg](https://mesonbuild.com/Reference-manual_functions.html#shared_library). If those are not specified, the default is to use the soname version for both (`soversion:` kwarg). If the `soversion:` kwarg is not specified, the first component of the project version is picked (and this trickles down to everything derived from `soversion`).

`-install_name` is always `libfoo.${soversion}.dylib`, same things as above apply.

**Comment by sdroege** [comment]
@lu-zero I guess let's get this in then and a new release with these changes would be useful to :)

**Comment by sdroege** [comment]
Let's get this in then? :)

**Pull request merged**